srtcore/core.cpp

```diff
@@ -603,33 +603,38 @@ void CUDT::setOpt(SRT_SOCKOPT optName, const void* optval, int optlen)
         break;
 
     case SRTO_TSBPDMODE:
-        if (m_bConnected)
-            throw CUDTException(MJ_NOTSUP, MN_ISCONNECTED, 0);
+        if (m_bOpened)
+            throw CUDTException(MJ_NOTSUP, MN_ISBOUND, 0);
+
         m_bOPT_TsbPd = cast_optval<bool>(optval, optlen);
         break;
 
     case SRTO_LATENCY:
-        if (m_bConnected)
-            throw CUDTException(MJ_NOTSUP, MN_ISCONNECTED, 0);
+        if (m_bOpened)
+            throw CUDTException(MJ_NOTSUP, MN_ISBOUND, 0);
```

Comment:
This change does not allow to set the latency after SRT connection is opened, right?
The listener connection is opened if `srt_bind(..)` was called or if `srt_listen(..)` is called, right?
Similar for the caller: either `srt_bind(..)` or `srt_connect(..)`.

It makes sense to have this in a separate commit, that changes the core library, not to mess it with application-level changes.

Reply:
No. The change does not allow the latency to be changed after binding. The latency never was allowed to be changed after connection.

The only difference is that earlier, if you wanted to bind the socket manually before connecting (instead of letting connect self-bind the socket), the latency was allowed to be altered AFTER binding, but before connecting.

Reply:
>earlier, if you wanted to bind the socket manually before connecting (instead of letting connect self-bind the socket), the latency was allowed to be altered AFTER binding, but before connecting.

Right. So this is a core change. Please split this PR into two commits using `git rebase`:
- [apps] Fixed srt-file-transmit: set transtype properly.
Added URI reassembling for UriParser. Fixed options copied during the opening to allow only before bind.
- [core] Do not allow to set socket option after bind